### PR TITLE
GH-1516 Documentation updates - TerraExecutor retry

### PR DESCRIPTION
**Pull request by okotsopoulos**
### Purpose

- https://broadinstitute.atlassian.net/browse/GH-1516

Documenting `/retry` behavior changes stemming from https://github.com/broadinstitute/wfl/pull/525

Namely, Terra submission ID is now a required filter, and workflow status is optional (and decorative).

### Review Instructions

Review is easiest off of a local docsite.  Per `docs/readme.md`:
```
make docs
. derived/.venv/docs/bin/activate && (  cd derived/docs; python3 -m mkdocs serve  )
```

Altered pages:
http://127.0.0.1:8000/executor/
http://127.0.0.1:8000/modules-covid/
http://127.0.0.1:8000/usage-retry/

Please read through and let me know if it makes sense.  I will merge following the v0.10.x release, when the corresponding functionality is live.

### CODE REVIEW

File: docs/md/modules-covid.md [hunk 1]
```diff
@@ -154,17 +154,18 @@ The response is an array of [workload objects](#workload-response-format).
 
 **GET /api/v1/workload/{uuid}/workflows**
 
-Query WFL for all workflows associated with workload `uuid`.
+Query WFL for all unretried workflows associated with workload `uuid`.
```
Comment (tbl3rd) [inline docs/md/modules-covid.md:157]: New word for me. ;-)

File: docs/md/modules-covid.md [hunk 2]
```diff
@@ -185,31 +186,35 @@ when available (when the workflow has succeeded).
     } ]
     ```
 
-**GET /api/v1/workload/{uuid}/workflows?status={status}**
+**GET /api/v1/workload/{uuid}/workflows?submission={submission}&status={status}**
 
-Query WFL for all workflows with `status` associated with workload `uuid`.
+Query WFL for all unretried workflows associated with workload `uuid`,
+filtering by any **optional** workflow filters specified as query params:
```
Comment (tbl3rd) [inline docs/md/modules-covid.md:192]: FWIW: Maybe `any` implies `optional`?

File: docs/md/usage-retry.md [hunk 3]
```diff
@@ -57,6 +70,10 @@ and cannot yet submit a subset of a snapshot.
 So retrying any workflow from a workload snapshot
 will resubmit all entities from that snapshot.
 
+(Because of this, the optional workflow status filter is purely decorative:
+all sibling workflows from the same submission will be resubmitted,
+regardless of their status.)
```
Comment (tbl3rd) [inline docs/md/usage-retry.md:75]: :-(

**Review by tbl3rd — APPROVED** [review]
Thanks.

File: docs/md/usage-retry.md (hunk 3 above)
Comment (okotsopoulos) [reply docs/md/usage-retry.md:75]: Ditto.

I thought about removing status entirely as a retry input for this reason, but ultimately we hope to use it when we can submit snapshot subsets.

I could still do this as a one-liner (plus documentation updates), by keeping the underlying implementation but removing it from the retry request body spec.

https://github.com/broadinstitute/wfl/blob/develop/api/src/wfl/api/spec.clj#L21

What do you think?

File: docs/md/modules-covid.md (hunk 2 above)
Comment (okotsopoulos) [reply docs/md/modules-covid.md:192]: Yes, I think you're right!

File: docs/md/usage-retry.md (hunk 3 above)
Comment (tbl3rd) [reply docs/md/usage-retry.md:75]: I say leave it as a "latent feature".

**Pull request merged**